### PR TITLE
ZH-trans: add kubeadm-upgrade-ha-1-12.md

### PR DESCRIPTION
**Pull request by xichengliudui**
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> Please delete this note before submitting the pull request.
>
> For 1.14 Features: set Milestone to 1.14 and Base Branch to dev-1.14
> 
> For Chinese localization, base branch to release-1.12
>
> For Korean Localization: set Base Branch to dev-1.13-ko.<latest team milestone>
>
> Help editing and submitting pull requests:
> https://kubernetes.io/docs/contribute/start/#improve-existing-content.
>
> Help choosing which branch to use:
> https://kubernetes.io/docs/contribute/start#choose-which-git-branch-to-use.
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>

### CODE REVIEW

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+### Upgrade the first control plane node
+
+Modify `configmap/kubeadm-config` for this control plane node:
+-->
+
+## 叠加控制平面节点
+
+### 升级第一个控制平面节点
+
+修改 `configmap/kubeadm-config` 控制平面节点：
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
为该控制平面节点修改 `configmap/kubeadm-config` 文件：
```

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+    This should be updated to the local node's IP address.       
+-->
+
+- `api.advertiseAddress`
+
+    应将其设置为本地节点的 IP 地址。
+
+- `etcd.local.extraArgs.advertise-client-urls`
+
+    这应该更新为本地节点的 IP 地址。
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
    此值应该更新为本地节点的 IP 地址。
```

下同。

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+### Upgrading additional control plane nodes
+
+Each additional control plane node requires modifications that are different from the first control plane node. Run:
+-->
+
+`kubeadm-config` ConfigMap 现在从 `v1alpha2` 版本更新为 `v1alpha3`。
+
+## 升级其他控制平面节点
+
+每个附加的控制平面节点都需要与第一个控制平面节点做不同的修改。运行：
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
每个额外的控制平面节点都需要与第一个控制平面节点做不同的修改。运行：
```

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+
+```shell
+kubectl get configmap -n kube-system kubeadm-config -o yaml > kubeadm-config-cm.yaml
+```
+
+<!--
+Open the file in an editor and replace the following values for `ClusterConfiguration`:
+-->
+在编辑器中打开文件并替换以下值 `ClusterConfiguration`：
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
在编辑器中打开文件并为 `ClusterConfiguration` 替换以下值：
```

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+### Upgrade each control plane
+
+Get a copy of the kubeadm config used to create this cluster. The config should be the same for every node. The config must exist on every control plane node before the upgrade begins.
+-->
+
+## 外部 etcd
+
+### 升级每个控制平面
+
+获取用于创建此群集的 kubeadm 配置的副本。每个节点的配置应该相同。在升级开始之前，配置必须存在于每个控制平面节点上。
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]: 群集 -> 集群

Comment (tengqm) [reply content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]: 每个 -> 所有

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+```
+
+<!--
+Open the file in an editor and set `api.advertiseAddress` to the local node's IP address.
+
+Now run the upgrade on each control plane node one at a time.
+-->
+在编辑器中打开文件并设置 `api.advertiseAddress` 为本地节点的 IP 地址。
+
+现在，一次在每个控制平面节点上运行升级。
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
现在，逐个控制平面节点地运行升级。
```

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 7]
```diff
@@ -0,0 +1,423 @@
...
+
+等待 etcd 进程重新连接。其他 etcd 节点日志中将出现错误警告。这是预料之中的。
+
+在其他 etcd 主机上重复此步骤。
+
+## 下一步
+
+### 手动升级您的 CNI 提供商
+
+您的容器网络接口（CNI）提供程序可能有自己的升级说明。检查插件页面找到您的 CNI 提供商，看看是否需要采取其他升级步骤。
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]: 插件页面的链接丢了

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 8]
```diff
@@ -0,0 +1,423 @@
...
+通过在每个节点上运行以下命令来升级 kubelet 和 kubectl：
+
+<!--
+# use your distro's package manager, e.g. 'apt-get' on Debian-based systems
+# for the versions stick to kubeadm's output (see above)
+-->
+
+```shell
+# 使用你的发行版软件包管理器，例如基于 Debian 系统上的 'apt-get' 
+# 对于版本，坚持 kubeadm 的输出(参见上文)
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
# 对于版本，基于 kubeadm 的输出来设置(参见上文)
```

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 9]
```diff
@@ -0,0 +1,423 @@
...
+apt-mark hold kubelet kubectl && \
+systemctl restart kubelet
+```
+
+<!--
+In this example a _deb_-based system is assumed and `apt-get` is used for installing the upgraded software. On rpm-based systems the command is `yum install <PACKAGE>=<NEW-K8S-VERSION>` for all packages.
+
+Verify that the new version of the kubelet is running:
+-->
+在本例中，假设一个基于 _deb_-based 的系统，并使用 `apt-get` 安装升级后的软件。在基于rpm的系统上，该命令 `yum install <PACKAGE>=<NEW-K8S-VERSION>` 适用于所有包。
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
本例假设使用一个基于 _deb_ 的系统，并使用 `apt-get` 安装升级后的软件。在基于 rpm 的系统上，对于所有软件包都使用 `yum install <PACKAGE>=<NEW-K8S-VERSION>` 命令。
```

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 10]
```diff
@@ -0,0 +1,423 @@
...
+验证新版本的 `kubelet` 是否正在运行：
+
+```shell
+systemctl status kubelet
+```
+
+<!--
+Verify that the upgraded node is available again by running the following command from wherever you run `kubectl`:
+-->
+通过从任何位置运行以下命令，验证已升级的节点是否可用 kubectl：
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
通过从 kubectl 所处的任何位置运行以下命令，再次验证已升级的节点是否可用：
```

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 11]
```diff
@@ -0,0 +1,423 @@
...
+
+```shell
+kubectl get nodes
+```
+
+<!--
+If the `STATUS` column shows `Ready` for the upgraded host, you can continue. You might need to repeat the command until the node shows `Ready`.
+-->
+如果 `STATUS` 列显示 `Ready` 升级后的主机，则可以继续。否则您可能需要重复该命令，直到节点显示 `Ready`。
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
如果对于升级后的主机其 `STATUS` 列显示 `Ready`，则可以继续；否则您可能需要重复该命令，直到节点显示 `Ready`。
```

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 12]
```diff
@@ -0,0 +1,423 @@
...
+   You can run `kubeadm upgrade apply` again, because it is idempotent and should eventually make sure the actual state is the desired state you are declaring. You can run `kubeadm upgrade apply` to change a running cluster with `x.x.x --> x.x.x` with `--force` to recover from a bad state.
+
+- If `kubeadm upgrade apply` on one of the secondary masters failed, the cluster is upgraded and working, but the secondary masters are in an undefined state. You need to investigate further and join the secondaries manually.
+-->
+
+## 如果出现问题
+
+如果升级失败，请查看是否适用以下方案之一：
+
+- 如果 `kubeadm upgrade apply` 无法升级群集，它将尝试执行回滚。如果在第一个主服务器上就是这种情况，则集群可能仍然完好无损。
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]: 群集 -> 集群

Comment (tengqm) [reply content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]: 主服务器 -> 主控节点

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 13]
```diff
@@ -0,0 +1,423 @@
...
+
+   您可以再次运行 `kubeadm upgrade apply`，因为它是幂等的，最终应该确保实际状态是您声明的期望状态。您可以运行 `kubeadm upgrade apply` 来使用 `x.x.x --> x.x.x` 更改正在运行的集群。`--force` 从错误状态恢复。
+
+- 如果一个代理主机上的 `kubeadm upgrade apply` 失败，集群将升级并运行，但是代理主机处于未定义的状态。您需要进一步研究并手动加入集群。
```
Comment (tengqm) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
- 如果其他节点上的 `kubeadm upgrade apply` 失败，则集群已经升级并运行，只是这些节点处于未定义的状态。您需要进一步调试并手动将其加入集群。
```

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 14]
```diff
@@ -0,0 +1,423 @@
...
+content_template: templates/task
+---
+-->
+
+{{% capture overview %}}
+
+<!--
+This page explains how to upgrade a highly available (HA) Kubernetes cluster created with `kubeadm` from version 1.11.x to version 1.12.x. In addition to upgrading, you must also follow the instructions in [Creating HA clusters with kubeadm](/docs/setup/independent/high-availability/).
+-->
+本页面解释如何从 1.11.x 版本升级使用 `kubeadm` 创建的高可用性（HA） Kubernetes 集群到版本 1.12.x。除了升级，您还必须遵循[使用 kubeadm 创建 HA 集群](/docs/setup/independent/high-availability/)。
```
Comment (SataQiu) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
本页介绍了如何将基于 kubeadm 创建的 Kubernetes HA 集群从 1.11.x 版本升级到 1.12.x 版本。除了升级，您还必须遵守[使用 kubeadm 创建 HA 集群](/docs/setup/independent/high-availability/) 的相关说明。
```

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 15]
```diff
@@ -0,0 +1,423 @@
...
+
+    应将其设置为本地节点的 IP 地址。
+
+- `etcd.local.extraArgs.advertise-client-urls`
+
+    此值应该更新为本地节点的 IP 地址。
+
+- `etcd.local.extraArgs.initial-advertise-peer-urls`
+
+   此值应该更新为本地节点的 IP 地址。
```
Comment (SataQiu) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
     此值应该更新为本地节点的 IP 地址。
```

Comment (SataQiu) [reply content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]: 图片

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 16]
```diff
@@ -0,0 +1,423 @@
...
+
+- `etcd.local.extraArgs.initial-cluster`
+
+<!--
+    This should be updated to include the hostname and IP address pairs for each control plane node in the cluster. For example:
+-->
+    应更新此项以包含群集中每个控制平面节点的主机名和 IP 地址对。例如：
+
+        "ip-172-31-92-42=https://172.31.92.42:2380,ip-172-31-89-186=https://172.31.89.186:2380,ip-172-31-90-42=https://172.31.90.42:2380"
+
```
Comment (SataQiu) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]: 图片

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 17]
```diff
@@ -0,0 +1,423 @@
...
+-->
+
+
+- `etcd.local.extraArgs.advertise-client-urls`
+
+    这应该更新为本地节点的 IP 地址。
+
+- `etcd.local.extraArgs.initial-advertise-peer-urls`
+
+   这应该更新为本地节点的 IP 地址。
```
Comment (SataQiu) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]: 图片

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 18]
```diff
@@ -0,0 +1,423 @@
...
+- `etcd.local.extraArgs.listen-peer-urls`
+
+    这应该更新为本地节点的 IP 地址。
+
+<!--
+You must also modify the `ClusterStatus` to add a mapping for the current host under apiEndpoints.
+
+Add an annotation for the cri-socket to the current node, for example to use docker:
+-->
+您还必须修改 `ClusterStatus` 为 apiEndpoints 下的当前主机添加映射。
```
Comment (SataQiu) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
您还必须修改 `ClusterStatus`， 为 apiEndpoints 下的当前主机添加映射。
```

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 19]
```diff
@@ -0,0 +1,423 @@
...
+
+将 cri-socket 的注释添加到当前节点，例如使用 docker：
```
Comment (SataQiu) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
将 cri-socket 的注解添加到当前节点，例如使用 docker：
```

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 20]
```diff
@@ -0,0 +1,423 @@
...
+```
+
+<!--
+Open the file in an editor and set `api.advertiseAddress` to the local node's IP address.
+
+Now run the upgrade on each control plane node one at a time.
+-->
+在编辑器中打开文件并设置 `api.advertiseAddress` 为本地节点的 IP 地址。
+
+现在，逐个控制平面节点地运行升级。
```
Comment (SataQiu) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
现在，在每个控制平面节点上分别执行升级命令。
```

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 21]
```diff
@@ -0,0 +1,423 @@
...
+验证新版本的 `kubelet` 是否正在运行：
+
+```shell
+systemctl status kubelet
+```
+
+<!--
+Verify that the upgraded node is available again by running the following command from wherever you run `kubectl`:
+-->
+通过从 kubectl 所处的任何位置运行以下命令，再次验证已升级的节点是否可用：
```
Comment (SataQiu) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
无论当前路径在哪里，使用 `kubectl` 运行以下命令，再次验证已升级的节点是否可用：
```

File: content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md [hunk 22]
```diff
@@ -0,0 +1,423 @@
...
+- If `kubeadm upgrade apply` failed to upgrade the cluster, it will try to perform a rollback. If this is the case on the first master, the cluster is probably still intact.
+
+   You can run `kubeadm upgrade apply` again, because it is idempotent and should eventually make sure the actual state is the desired state you are declaring. You can run `kubeadm upgrade apply` to change a running cluster with `x.x.x --> x.x.x` with `--force` to recover from a bad state.
+
+- If `kubeadm upgrade apply` on one of the secondary masters failed, the cluster is upgraded and working, but the secondary masters are in an undefined state. You need to investigate further and join the secondaries manually.
+-->
+
+## 如果出现问题
+
+如果升级失败，请查看是否适用以下方案之一：
```
Comment (SataQiu) [inline content/zh/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-12.md]:
```suggestion
如果升级失败，请检查是否符合以下列出的可能场景：
```

**Comment by fleeto** [comment]
`[trans-bot:N/A]`

Thank you @xichengliudui, the [related task issue(https://github.com/k8smeetup/website-tasks/issues/1434)] has no assignee. will not be reported to the task issues

**Comment by fleeto** [comment]
`[trans-bot:N/A]`

Thank you @xichengliudui, the [related task issue(https://github.com/k8smeetup/website-tasks/issues/1801)] has no assignee. will not be reported to the task issues

**Comment by fleeto** [comment]
`[trans-bot:N/A]`

Thank you @xichengliudui, the [related task issue(https://github.com/k8smeetup/website-tasks/issues/1370)] has no assignee. will not be reported to the task issues

**Comment by fleeto** [comment]
`[trans-bot:N/A]`

Thank you @xichengliudui, the [related task issue](https://github.com/k8smeetup/website-tasks/issues/1434) has no assignee. will not be reported to the task issues

**Pull request merged**